src/server/engine_shard_set.cc

```diff
@@ -137,8 +145,19 @@ bool EngineShard::DoDefrag() {
   const float threshold = GetFlag(FLAGS_mem_defrag_page_utilization_threshold);
 
   auto& slice = db_slice();
-  DCHECK(slice.IsDbValid(kDefaultDbIndex));
-  auto [prime_table, expire_table] = slice.GetTables(kDefaultDbIndex);
+
+  // If we moved to an invalid db, skip as long as it's not the last one
+  if (!slice.IsDbValid(defrag_state_.dbid) && defrag_state_.dbid + 1 < slice.db_array_size())
```

Comment:
I don't understand why we're increasing dbid only once. Should this be a loop or something? What would happen if two consecutive DB IDs are invalid?

Reply:
True that somehow became an if even though the comment clearly is phrased as if its a loop 🙂